astropy/io/ascii/cparser.pyx

```diff
@@ -876,13 +883,20 @@ def _read_chunk(CParser self, start, end, try_int,
         delete_tokenizer(chunk_tokenizer)
         queue.pop()
         queue.put((None, e, i))
+        return
 
     reconvert_cols = reconvert_queue.get()
     for col in reconvert_cols:
         queue.put((self._convert_str(chunk_tokenizer, col, -1), i, col))
     delete_tokenizer(chunk_tokenizer)
     reconvert_queue.put(reconvert_cols) # return to the queue for other processes
 
+    # Since we already deallocated the tokenizer above, we should make sure we
+    # set self.tokenizer to NULL to avoid deallocating the tokenizer again
+    # inside __dealloc__.
+    self.tokenizer = NULL
```

Comment:
This may be needed in the two except clauses above too ?

Reply:
Fixed